File: src/Suframa.Sciex.CrossCutting.DataTransferObject/ViewModel/EstruturaPropriaPLIVM.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Suframa.Sciex.CrossCutting.DataTransferObject.ViewModel
{
	public class EstruturaPropriaPLIVM : PagedOptions
	{
		public long? IdEstruturaPropria { get; set; }
		public int? NumeroProtocolo { get; set; }
		public DateTime DataEnvio { get; set; }
		public DateTime? DataInicioProcessamento { get; set; }
		public DateTime? DataFimProcessamento { get; set; }
		public byte? StatusProcessamentoArquivo { get; set; }
		public short? QuantidadePLIArquivo { get; set; }
		public short? QuantidadePLIProcessadoSucesso { get; set; }
		public short? QuantidadePLIProcessadoFalha { get; set; }
		public string VersaoEstrutura { get; set; }
		public string LoginUsuarioEnvio { get; set; }
		public string NomeUsuarioEnvio { get; set; }
		public string CNPJImportador { get; set; }
		public string RazaoSocialImportador { get; set; }
		public int? InscricaoCadastral { get; set; }
		public string NomeArquivoEnvio { get; set; }
		public string ListaDePLI { get; set; }
		public byte? StatusPLITecnologiaAssistiva { get; set; }
		public string DescricaoPendenciaImportador { get; set; }
		public byte StatusUltimoArquivoProcessamento { get; set; }
		public decimal TipoArquivo { get; set; }

		// Complemento de Classe
		public DateTime? DataInicio { get; set; }
		public DateTime? DataFim { get; set; }
		public List<SolicitacaoPliVM> ListaSolicitacao { get; set; }
		public string QuantidadePliConcatenado { get; set; }
		public string StatusValidacaoArquivoConcatenado { get; set; }

	}

}

File: src/Suframa.Sciex.UserInterface/www/app/view-model/EstruturaPropriaPLIVM.ts
import { SolicitacaoPliVM } from './SolicitacaoPLIVM';

export class EstruturaPropriaPLIVM {

	idEstruturaPropria?: number;
	dataEnvio: Date;
	dataInicioProcessamento?: Date;
	dataFimProcessamento?: Date;
	statusProcessamentoArquivo?: number;
	quantidadePLIArquivo?: number;
	quantidadePLIProcessadoSucesso?: number;
	quantidadePLIProcessadoFalha?: number;
	versaoEstrutura: string;
	loginUsuarioEnvio: string;
	nomeUsuarioEnvio: string;
	CNPJImportador: string;
	razaoSocialImportador: string;
	inscricaoCadastral?: number
	nomeArquivoEnvio: number;
	listaDePLI: string;
	statusPLITecnologiaAssistiva?: number;
	descricaoPendenciaImportador: string;
	statusUltimoArquivoProcessamento: number;

	// Complemento de Classe
	dataInicio?: Date;
	dataFim?: Date;
	listaSolicitacao: Array<SolicitacaoPliVM>;
	quantidadePliConcatenado: string
	statusValidacaoArquivoConcatenado: string;	

}
